aries_cloudagent/core/event_bus.py

```diff
@@ -0,0 +1,119 @@
+        return self._payload
+
+    def __eq__(self, other):
+        """Test equality."""
+        if not isinstance(other, Event):
+            return False
+        return self._topic == other._topic and self._payload == other._payload
+
+    def __repr__(self):
+        """Return debug representation."""
+        return "<Event topic={}, payload={}>".format(self._topic, self._payload)
+
+
+class EventBus:
+    """A simple event bus implementation."""
+
+    def __init__(self):
+        """Initialize Event Bus."""
+        self.topic_patterns_to_subscribers: Dict[Pattern, Sequence[Callable]] = {}
+
+    async def notify(self, profile: "Profile", event: Event):
+        """Notify subscribers of event.
+
+        Args:
+            profile (Profile): context of the event
+            event (Event): event to emit
+
+        """
+        # TODO don't block notifier until subscribers have all been called?
+        # TODO trigger each processor but don't await?
+        # TODO log errors but otherwise ignore?
+
+        LOGGER.debug("Notifying subscribers: %s", event)
+        matched = [
+            processor
+            for pattern, processor in self.topic_patterns_to_subscribers.items()
+            if pattern.match(event.topic)
+        ]
+
+        for processor in chain(*matched):
+            try:
+                await processor(profile, event)
+            except Exception:
+                LOGGER.exception("Error occurred while processing event")
+
+    def subscribe(self, pattern: Pattern, processor: Callable):
+        """Subscribe to an event.
+
+        Args:
+            pattern (Pattern): compiled regular expression for matching topics
+            processor (Callable): async callable accepting profile and event
+
+        """
+        LOGGER.debug("Subscribed: topic %s, processor %s", pattern, processor)
+        if pattern not in self.topic_patterns_to_subscribers:
+            self.topic_patterns_to_subscribers[pattern] = []
+        self.topic_patterns_to_subscribers[pattern].append(processor)
+
+    def unsubscribe(self, pattern: Pattern, processor: Callable):
+        """Unsubscribe from an event.
+
+        This method is idempotent. Repeated calls to unsubscribe will not
+        result in errors.
+
+        Args:
+            pattern (Pattern): regular expression used to subscribe the processor
+            processor (Callable): processor to unsubscribe
+
+        """
+        if pattern in self.topic_patterns_to_subscribers:
+            try:
+                index = self.topic_patterns_to_subscribers[pattern].index(processor)
+            except ValueError:
+                return
+            del self.topic_patterns_to_subscribers[pattern][index]
+            if not self.topic_patterns_to_subscribers[pattern]:
+                del self.topic_patterns_to_subscribers[pattern]
+            LOGGER.debug("Unsubscribed: topic %s, processor %s", pattern, processor)
+
+
+class MockEventBus(EventBus):
```

Comment:
It is a test class for test purposes that only is used on unit tests.
This shouldn't be declared here, isn't it?

Reply:
This follows a pattern used elsewhere in ACA-Py so I don't see a problem with defining it here. This makes it easier to reuse in tests.